components/UnitCsvViewer.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div v-if="loading">Loading</div>
+  <v-data-table v-else v-bind="{ ...csvContent }" multi-sort />
+</template>
+
+<script>
+import * as csv from '@fast-csv/parse'
+
+export default {
+  name: 'UnitCsvViewer',
+  props: {
+    file: {
+      type: File,
+      required: true
+    },
+    preprocessorFunc: {
+      type: Function,
+      default: () => a => a
+    }
+  },
+  data() {
+    return {
+      csvText: '',
+      csvContent: {},
+      loading: true
+    }
+  },
+  watch: {
+    async file(newFile) {
+      await this.getContentFromFile(newFile)
+    }
+  },
+  async created() {
```

Comment:
We can use use `immediate: true` in the `watch` property above, instead of defining a `created` property that does the same thing.

Reply:
That's exactly what I needed, I'll change it!

Reply:
Done, thanks!